llvm/lib/Target/AIE/AIEBaseSubtarget.cpp

```diff
@@ -482,6 +508,26 @@ class WAWEdges : public ScheduleDAGMutation {
   };
 };
 
+// Adds WAW edges for scheduling in the context of the Scheduler.
+// This class extends WAWEdges to apply WAW edges using a Scheduler if available
+// It overrides the apply method to retrieve the Scheduler from the DAG if a
+// BasicBlock is present, otherwise, it uses nullptr.
+class SchedWAWEdges : public WAWEdges {
```

Comment:
Nit: Maybe name this `MachineSchedWAWEdges` to be a bit more explicit. The `SWP` is also a scheduler.